### DISCUSSION TITLE
Live coding platform

### DISCUSSION THREAD

**Discussion by BFergerson** [apache/skywalking#7864] [Show and tell]
Hello SkyWalkers,

I would like to introduce you to [Source++](https://github.com/sourceplusplus/live-platform), an open-source live coding platform. Powered by Apache SkyWalking, Source++ enhances the software development experience by allowing you to add breakpoints, logs, metrics, and distributed tracing to live production software in real-time on-demand, right from your IDE or CLI.

I'd appreciate any feedback the SkyWalking community can offer. Source++ only works with JVM languages, but Python support should be less than a few weeks away. If you have trouble installing Source++, please make a ticket. I'll improve the documentation.

<details>
  <summary>Architecture</summary>
  
  ![](https://raw.githubusercontent.com/sourceplusplus/live-platform/master/.github/media/sourcepp_architecture.jpg)
</details>

### Live Breakpoint

> Live Breakpoints (a.k.a non-breaking breakpoints) are useful debugging instruments for gaining insight into the live variables available in production at a given scope.

<details>
  <summary>Explanation Diagram</summary>
  
  ![bp](https://user-images.githubusercontent.com/3278877/136123227-f8427c30-65dc-47a4-82d2-a395cbad3afc.png)
</details>

<details>
  <summary>Screencast</summary>

  ![live-breakpoint](https://user-images.githubusercontent.com/3278877/136304451-2c98ad30-032b-4ce0-9f37-f98cd750adb3.gif)
</details>

### Live Log

> Live Logs (a.k.a just-in-time logging) are quick and easy debugging instruments for instantly outputting live data from production without redeploying or restarting your application.

<details>
  <summary>Explanation Diagram</summary>
  
  ![log](https://user-images.githubusercontent.com/3278877/136123237-b8251ec9-c1eb-4beb-99e5-1ab114efcb28.png)
</details>

<details>
  <summary>Screencast</summary>

  ![live-log](https://user-images.githubusercontent.com/3278877/136304738-d46c2796-4dd3-45a3-81bb-5692547c1c71.gif)  
</details>

**Comment by wu-sheng** [comment]
Super cool, this is for you.

https://twitter.com/asfskywalking/status/1446089511855747073?s=21
